### PR TITLE
croc: manually update to 10.0.8

### PR DESCRIPTION
**Pull request by jonasjelonek**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

Although croc is included in auto-bump, no updates have been made since three weeks despite having multiple upstream updates. I suspect that the last update to 10.0.4, which was mistakenly released as 10.04 from the upstream author in the first place, broke the auto-bump.

### CODE REVIEW

**Comment by jonasjelonek** [comment]
CI fails, checking the failing command locally produces:
```bash
# brew audit --formula croc --online --git --skip-style
croc
  * stable version should not decrease (from 10.04 to 10.0.8)
Error: 1 problem in 1 formula detected.
```

I guess this supports that the auto-bump is broken due to the version issue.

**Comment by jonasjelonek** [comment]
It seems like the testing fails since passing the secret via CLI args was also disabled on macOS after 10.0.4. I didn't recognize this when running brew test because I didn't test it on macOS. I'll try to fix this.

**Comment by jonasjelonek** [comment]
@chenrui333 failing again due to an audit issue related to the version

**Review by chenrui333 — APPROVED** [review]

File: Formula/c/croc.rb [hunk 1]
```diff
@@ -24,7 +24,7 @@ def install
 
   test do
     # As of https://github.com/schollz/croc/pull/701 an alternate method is used to provide the secret code
-    ENV["CROC_SECRET"] = "homebrew-test" if OS.linux?
+    ENV["CROC_SECRET"] = "homebrew-test" if OS.linux? || OS.mac?
```
Comment (ZhongRuoyu) [inline Formula/c/croc.rb:27]:
```suggestion
    ENV["CROC_SECRET"] = "homebrew-test"
```

**Review by ZhongRuoyu — CHANGES_REQUESTED** [review]

File: Formula/c/croc.rb (hunk 1 above)
Comment (jonasjelonek) [reply Formula/c/croc.rb:27]: that's actually better, since Homebrew is only for Linux and macOS. Thanks.

**Review by ZhongRuoyu — APPROVED** [review]
Thanks for spotting this @jonasjelonek!

**Review by chenrui333 — APPROVED** [review]

**Pull request merged**